### PR TITLE
Added missing fields in ServerHardwareType Resource

### PR DESCRIPTION
**Pull request by AnushaY1916**
### Description
Added missing fields in ServerHardwareType Resource

### Issues Resolved
No

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass for go 1.11 + gofmt checks.
- [ ] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [ ] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: ov/server_hardware_type.go [hunk 1]
```diff
@@ -9,15 +9,95 @@ import (
...
 
+type VirtualPort struct {
+	Capabilities []string `json:"capabilities,omitempty"` // "capabilities": {}
+	PortFunction string   `json:"portFunction,omitempty"` // "portFunction": "a"
+	PortNumber   int      `json:"portNumber,omitempty"`   // "portNumber": 2
+}
+type SHTPort struct {
+	Mapping               int           `json:"mapping,omitempty"`               // "mapping": 3
+	MaxSpeedMbps          int           `json:"maxSpeedMbps,omitempty"`          // "maxSpeedMbps":3553
+	maxVFsSupported       int           `json:"maxVFsSupported,omitempty"`       // "maxVFsSupported": 23
```
Comment (madhav-bharadwaj) [inline ov/server_hardware_type.go]: Can you change the case here?

File: ov/server_hardware_type.go [hunk 2]
```diff
@@ -9,15 +9,95 @@ import (
...
+type Adapter struct {
+	Capabilities    []string  `json:"capabilities,omitempty"`    // "capabilities": {}
+	DeviceNumber    int       `json:"deviceNumber,omitempty"`    // "deviceNumber": 123
+	DeviceType      string    `json:"deviceType,omitempty"`      // "deviceType":"Ethernet"
+	Location        string    `json:"location,omitempty"`        // "location": "Mezz"
+	MaxVFsSupported int       `json:"maxVFsSupported,omitempty"` // "maxVFsSupported": 256
+	MinVFsIncrement int       `json:"minVFsIncrement,omitempty"` // "minVFsIncrement": 8
+	Model           string    `json:"model,omitempty"`           // "model": ""
+	Ports           []SHTPort `json:"ports,omitempty"`           // "ports": {}
+	Slot            int       `json:"slot,omitempty"`            // "slot": 3
```
Comment (madhav-bharadwaj) [inline ov/server_hardware_type.go]: `StorageCapabilities` is part of this attribute as well. Can you check?

Comment (AnushaY1916) [reply ov/server_hardware_type.go]: I missed it in Adapter struct, added it now

File: ov/server_hardware_type.go (hunk 1 above)
Comment (AnushaY1916) [reply ov/server_hardware_type.go]: Yes, changed it.

**Pull request merged**